custom_components/econet300/number.py

```diff
@@ -129,10 +129,17 @@ async def async_set_native_value(self, value: float) -> None:
 
 def can_add(key: str, coordinator: EconetDataCoordinator):
     """Check if a given entity can be added based on the availability of data in the coordinator."""
-    return coordinator.has_param_edit_data(key) and coordinator.data["paramsEdits"][key]
+    try:
```

Comment:
Were there any problems with this ?

Reply:
I thought is better because it adds robustness by handling potential KeyError exceptions, ensuring that missing keys do not cause the function to fail unexpectedly. I can revert back old version if it better

Reply:
I this hsandlinjg add aditional value to error tracking and debuging then you can leave it